addons/survey/models/survey_survey.py

```diff
@@ -563,28 +588,52 @@ def _session_trigger_next_question(self):
             return
 
         if not self.session_question_id:
-            question = self.question_ids[0]
+            next_question = self.question_ids[0]
         else:
-            questions = list(enumerate(self.question_ids))
-            current_question_index = questions.index(
-                next(question for question in questions if question[1] == self.session_question_id)
-            )
-            if current_question_index < len(self.question_ids) - 1:
-                question = self.question_ids[current_question_index + 1]
-            else:
-                question = self.session_question_id
-
-        # using datetime.datetime because we want the millis portion
-        now = datetime.datetime.now()
-        self.sudo().write({
-            'session_question_id': question.id,
-            'session_question_start_time': fields.Datetime.now() + relativedelta(seconds=2)
-        })
-        self.env['bus.bus'].sendone(self.access_token, {
-            'question_start': now.timestamp(),
-            'type': 'next_question'
+            most_voted_answers = self._get_session_most_voted_answers()
+            next_question = self._get_next_question(most_voted_answers, self.session_question_id.id)
+
+        return next_question
+
+    def _get_session_most_voted_answers(self):
```

Comment:
In fact, we need the previous answers because it can be that a previous answer triggers the next question, espacially if the current question is not triggering another question.